legend-pure-m3-core/src/main/java/org/finos/legend/pure/m3/serialization/runtime/Source.java

```diff
@@ -453,6 +589,68 @@ public ConcreteFunctionDefinition<?> findConcreteFunctionDefinitionAt(int line,
         }
     }
 
+    public SourceCoordinates.Preview getPreviewTextWithCoordinates(int startLine, int startColumn, int endLine, int endColumn)
+    {
+        if (StringIterate.isEmpty(this.content))
+        {
+            return null;
+        }
+
+        Matcher linesMatcher = LINE_PATTERN.matcher(this.content);
+        List<String> lines = new ArrayList<>();
+        while (linesMatcher.find())
+        {
+            lines.add(linesMatcher.group());
+        }
```

Comment:
If you're going to get all the lines, why not use String.split("\\R")? The only reason to use that LINE_MATCHER is to avoid going through all the lines of the file.